src/EnergyPlus/WindowManager.cc

```diff
@@ -3842,31 +3841,34 @@ namespace WindowManager {
             } else if (ANY_SHADE_SCREEN(ShadeFlag)) {
-                TransDiff = state.dataConstruction->Construct(ConstrNumSh).TransDiff;
+                reflDiff = state.dataConstruction->Construct(ConstrNumSh).ReflectSolDiffBack;
             } else if (ANY_BLIND(ShadeFlag)) {
                 if (state.dataSurface->SurfWinMovableSlats(SurfNum)) {
-                    TransDiff =
-                        General::Interp(state.dataConstruction->Construct(ConstrNumSh).BlTransDiff(state.dataSurface->SurfWinSlatsAngIndex(SurfNum)),
-                                        state.dataConstruction->Construct(ConstrNumSh)
-                                            .BlTransDiff(std::min(Material::MaxSlatAngs, state.dataSurface->SurfWinSlatsAngIndex(SurfNum) + 1)),
-                                        state.dataSurface->SurfWinSlatsAngInterpFac(SurfNum));
+                    reflDiff = General::Interp(
+                        state.dataConstruction->Construct(ConstrNumSh).BlReflectSolDiffBack(state.dataSurface->SurfWinSlatsAngIndex(SurfNum)),
+                        state.dataConstruction->Construct(ConstrNumSh)
+                            .BlTransDiff(std::min(Material::MaxSlatAngs, state.dataSurface->SurfWinSlatsAngIndex(SurfNum) + 1)),
+                        state.dataSurface->SurfWinSlatsAngInterpFac(SurfNum));
                 } else {
-                    TransDiff = state.dataConstruction->Construct(ConstrNumSh).BlTransDiff(1);
+                    reflDiff = state.dataConstruction->Construct(ConstrNumSh).BlReflectSolDiffBack(1);
                 }
             } else if (ShadeFlag == WinShadingType::SwitchableGlazing) {
-                TransDiff = InterpSw(state.dataSurface->SurfWinSwitchingFactor(SurfNum),
-                                     state.dataConstruction->Construct(ConstrNum).TransDiff,
-                                     state.dataConstruction->Construct(ConstrNumSh).TransDiff);
+                reflDiff = InterpSw(state.dataSurface->SurfWinSwitchingFactor(SurfNum),
+                                    state.dataConstruction->Construct(ConstrNum).ReflectSolDiffBack,
+                                    state.dataConstruction->Construct(ConstrNumSh).ReflectSolDiffBack);
             }
             // shouldn't this be + outward flowing fraction of absorbed SW? -- do not know whose comment this is?  LKL (9/2012)
             state.dataSurface->SurfWinLossSWZoneToOutWinRep(SurfNum) =
                 state.dataHeatBal->EnclSolQSWRad(state.dataSurface->Surface(SurfNum).SolarEnclIndex) * state.dataSurface->Surface(SurfNum).Area *
-                TransDiff;
-            state.dataSurface->SurfWinHeatGain(SurfNum) -= state.dataSurface->SurfWinLossSWZoneToOutWinRep(SurfNum);
+                    (1 - reflDiff) +
+                state.dataHeatBalSurf->SurfWinInitialBeamSolInTrans(SurfNum);
+            state.dataSurface->SurfWinHeatGain(SurfNum) -=
```

Comment:
Subtract missing solar leaving components from `SurfWinHeatGain` for regular windows.